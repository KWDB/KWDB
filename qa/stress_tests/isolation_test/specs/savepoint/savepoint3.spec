setup
{
  CREATE TABLE foo (
     key INT PRIMARY KEY,
     value INT
  );

  INSERT INTO foo VALUES (1, 1);
}

teardown
{
  DROP TABLE foo;
}

session s1
step s1			{ BEGIN transaction isolation level read committed; }
step s1l		{ SELECT * FROM foo; }
step s1svp		{ SAVEPOINT f; }
step s1d		{ DELETE FROM foo WHERE TRUE; }
step s1r		{ ROLLBACK TO f; }
step s1c		{ COMMIT; }

session s2
step s2			{ BEGIN transaction isolation level read committed; }
step s2l		{ SELECT * FROM foo FOR UPDATE; }
step s2c		{ COMMIT; }

permutation s1 s2 s1l s1svp s1d s1r s1c s2l s2c
permutation s1 s2 s1l s1svp s1d s1r s2l s1c s2c
permutation s1 s2 s1l s1svp s1d s2l s1r s1c s2c
permutation s1 s2 s2l s1l s2c s1svp s1d s1r s1c
permutation s1 s2 s2l s2c s1l s1svp s1d s1r s1c
