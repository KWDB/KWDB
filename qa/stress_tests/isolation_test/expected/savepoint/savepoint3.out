Parsed test spec with 2 sessions

starting permutation: s1 s2 s1l s1svp s1d s1r s1c s2l s2c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1l: SELECT * FROM foo;  finished.
key|value
---+-----
  1|    1
(1 row)

step s1svp: SAVEPOINT f;  finished.
step s1d: DELETE FROM foo WHERE TRUE;  finished.
step s1r: ROLLBACK TO f;  finished.
step s1c: COMMIT;  finished.
step s2l: SELECT * FROM foo FOR UPDATE;  finished.
key|value
---+-----
  1|    1
(1 row)

step s2c: COMMIT;  finished.

starting permutation: s1 s2 s1l s1svp s1d s1r s2l s1c s2c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1l: SELECT * FROM foo;  finished.
key|value
---+-----
  1|    1
(1 row)

step s1svp: SAVEPOINT f;  finished.
step s1d: DELETE FROM foo WHERE TRUE;  finished.
step s1r: ROLLBACK TO f;  finished.
step s2l: SELECT * FROM foo FOR UPDATE; <waiting ...>
step s1c: COMMIT;  finished.
step s2l: <... completed>
key|value
---+-----
  1|    1
(1 row)

step s2c: COMMIT;  finished.

starting permutation: s1 s2 s1l s1svp s1d s2l s1r s1c s2c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s1l: SELECT * FROM foo;  finished.
key|value
---+-----
  1|    1
(1 row)

step s1svp: SAVEPOINT f;  finished.
step s1d: DELETE FROM foo WHERE TRUE;  finished.
step s2l: SELECT * FROM foo FOR UPDATE; <waiting ...>
step s1r: ROLLBACK TO f;  finished.
step s1c: COMMIT;  finished.
step s2l: <... completed>
key|value
---+-----
  1|    1
(1 row)

step s2c: COMMIT;  finished.

starting permutation: s1 s2 s2l s1l s2c s1svp s1d s1r s1c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s2l: SELECT * FROM foo FOR UPDATE;  finished.
key|value
---+-----
  1|    1
(1 row)

step s1l: SELECT * FROM foo;  finished.
key|value
---+-----
  1|    1
(1 row)

step s2c: COMMIT;  finished.
step s1svp: SAVEPOINT f;  finished.
step s1d: DELETE FROM foo WHERE TRUE;  finished.
step s1r: ROLLBACK TO f;  finished.
step s1c: COMMIT;  finished.

starting permutation: s1 s2 s2l s2c s1l s1svp s1d s1r s1c
step s1: BEGIN transaction isolation level read committed;  finished.
step s2: BEGIN transaction isolation level read committed;  finished.
step s2l: SELECT * FROM foo FOR UPDATE;  finished.
key|value
---+-----
  1|    1
(1 row)

step s2c: COMMIT;  finished.
step s1l: SELECT * FROM foo;  finished.
key|value
---+-----
  1|    1
(1 row)

step s1svp: SAVEPOINT f;  finished.
step s1d: DELETE FROM foo WHERE TRUE;  finished.
step s1r: ROLLBACK TO f;  finished.
step s1c: COMMIT;  finished.
